synapse/app/admin_cmd.py

```diff
@@ -232,11 +229,10 @@ def start(config_options):
     # We also make sure that `_base.start` gets run before we actually run the
     # command.
 
-    @defer.inlineCallbacks
     def run(_reactor):
         with LoggingContext("command"):
-            yield _base.start(ss, [])
-            yield args.func(ss, args)
+            _base.start(ss, [])
+            return defer.ensureDeferred(args.func(ss, args))
```

Comment:
This is not equivalent, as we'll exit the `with` block without awaiting on the deferred.

Reply:
Right. Unfortunately this one seems difficult to get rid of then...

Reply:
Luckily I don't think it is important to clear them out of this file. I've rolled back the changes here for now though.

Reply:
The alternative would to change this to async await and then do `task.react(lambda: defer.ensureDeferred(run())` or whatever below it.

Reply:
Yeah, that would work too, but feels kind of silly to do that. For the admin command, I don't think it is a big deal. Hopefully Twisted will one day support async functions for `task.run`?

Reply:
I should note that `ensureDeferred` essentially makes an `inlineCallback` anyway, so seems circular to do that.

Reply:
Yeah, I think the only reason to do it would be to get rid of all instances of `inlineCallback` in the code base as a matter of sanity and cleanliness

Reply:
Fair. I'll make the change. 👍